ttnn/cpp/ttnn/graph/graph_query_op_constraints.hpp

```diff
@@ -44,30 +44,30 @@ struct QueryResponse {
  */
 template <typename Op, typename... Args>
 auto query_op_constraints(Op op, Device* device, Args&&... args) {
-    uint32_t num_of_active_graph_captures = 0;
     try {
+        nlohmann::json op_trace;
         // outer graph capture is to avoid dispatching/allocating dummy input tensors
-        GraphProcessor::begin_graph_capture(GraphProcessor::RunMode::NO_DISPATCH);
-        num_of_active_graph_captures++;
+        {
+            auto capture_outer = GraphCaptureScopeGuard(GraphProcessor::RunMode::NO_DISPATCH);
```

Comment:
Interesting approach, I thought `GraphProcessor::begin_graph_capture` will return a guard instead. 
One concern I have is that `GraphCaptureScopeGuard` does not make it clear that you just started tracing.
It is not a major one. Wonder what @dmakoviichuk-tt thinks.

Reply:
It sounds ok for me. But btw if exception was throw it means we didn;t close. some functions. is it ok?

Reply:
Maybe changing the name to GraphCapture, ScopedGraphCapture, or ScopeGuardedGraphCapture might help. Thoughts?